hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/TestKMS.java

```diff
@@ -550,7 +550,7 @@ public Void call() throws Exception {
           threadGroup.enumerate(threads);
```

Comment:
Thanks @aajisaka  for reporting the issue and the fix. 

I think the problem is Line 550: ThreadGroup#enumerate does not provide a reliable way to get all the threads based on the JDK document. This is problematic especially if the threads array is not big enough to hold all the threads returned, the extra will be silently skipped. In the case of the reloader thread is skipped, we will have a reloaderThead as null for NPE. The proposed fix may not completely solve the problem. 

I would suggest we use Apache common ThreadUtils.findThreadsByName to replace the code Line 547-563, which simplify the code with better handling of this case to avoid NPE. 

```suggestion
 Collection<Thread> result =
              ThreadUtils.findThreadsByName(SSL_RELOADER_THREAD_NAME);
          Assert.assertEquals(1, result.size());
          Assert.assertTrue("Reloader is not alive",
              result.iterator().next().isAlive());
```

Reply:
Thanks @xiaoyuyao for your suggestion.
Updated to use `ThreadUtils.findThreadsByName`.

> `Assert.assertEquals(1, result.size());`

Actually, the size is 2. I commented why it is.

Reply:
Thanks @aajisaka  for the update. The change LGTM. You may also call ThreadUtils.findThreadsByName again and verify that the size is reduce by 1 compare with before. But I'm OK with the patch as-is.

Reply:
Thank you. Updated.